ledger-service/http-json-perf/src/main/scala/com/daml/http/perf/scenario/AsyncQueryConstantAcs.scala

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.http.perf.scenario
+
+import io.gatling.core.Predef._
+import io.gatling.http.Predef._
+import io.gatling.http.check.ws.WsTextFrameCheck
+
+import scala.concurrent.duration._
+
+@SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements"))
+class AsyncQueryConstantAcs
+    extends Simulation
+    with SimulationConfig
+    with HasRandomAmount
+    with HasCreateRequest {
+
+  private val wantedAcsSize = 5000
+
+  private val waitForResponse: FiniteDuration = 5.seconds
+
+  private val numberOfRuns = 100
+
+  private val queryRequest =
+    """{"templateIds": ["Iou:Iou"], "query": {"amount": {"%gt": 1.0}}}"""
+
+  val messageCheck: WsTextFrameCheck = ws
+    .checkTextMessage("messageCheck")
+    .check(jsonPath("$.offset").find.notExists)
+    .check(jsonPath("$.events[*].created").findAll)
+
+  private def query(runId: Int) = {
+    val wsName = s"websocket$runId"
+    ws("Connect websocket", wsName)
+      .connect("/v1/stream/query")
+      .subprotocol(s"jwt.token.$aliceJwt, daml.ws.auth")
+      .onConnected(
+        exec(
+          ws(s"Send Query Request and wait for response: $waitForResponse", wsName)
+            .sendText(queryRequest)
+            .await(waitForResponse)(Vector.fill(5)(messageCheck): _*)
```

Comment:
Just to make sure I’m reading this correctly: This means we’re benchmarking, establishing a connection and receiving five messages?

Reply:
yes